### PR TITLE
Create Student Signup Form

### PR DESCRIPTION
**Pull request by evantey14**
Made a basic student signup form with username, password, phone number and school. Users should be able to click Sign Up from the nav bar, fill out the form, and create a user on submit (then get logged in and redirected to the dashboard).

In hopes of keeping this small I've left the following as next steps:
* handle errors: e.g. username already exists / other malformed input errors
* fill in other fields
* make teacher signup
* do field validation: e.g make some fields required.

@mvadari in particular, let me know if there's code I can delete in order to clean up this PR

(edited to add detail)

### CODE REVIEW

**Comment by evantey14** [comment]
aha! figured out how to insert a screenshot
screenshot

File: client/src/accounts/StudentSignupForm.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  handleChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    this.setState((prevState: State) => {
+      const newState = { ...prevState };
+      if (isValidField(name, prevState)) {
+        newState[name] = value;
+      }
+      return newState;
+    });
```
Comment (wishdasher) [inline client/src/accounts/StudentSignupForm.tsx:42]: Can you add a `TODO(sophie): look more at this`?

File: client/src/accounts/StudentSignupForm.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        localStorage.setItem("refresh", result.data.tokens.refresh);
+        this.props.onLogin({ loggedIn: true });
+        navigate("dashboard");
+      });
+  };
+
+  render() {
+    return (
+      <form onSubmit={this.handleSignup}>
+        {renderLabeledInput(this.handleChange, "Username", "username", this.state.username, "text")}
```
[the remark this answers is not in the document]
Comment (evantey14) [reply client/src/accounts/StudentSignupForm.tsx:70]: lmao i feel like the minimalist in me does not like adding nonfunctional things to frontend.

alternatively we can see this as a *fun starter task opportunity* for someone who does enjoy this more than me.

File: client/src/accounts/StudentSignupForm.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          this.state.password,
+          "password"
+        )}
+        {renderLabeledInput(
+          this.handleChange,
+          "Phone Number",
+          "phoneNumber",
+          this.state.phoneNumber,
+          "tel"
+        )}
```
[the remark this answers is not in the document]
Comment (evantey14) [reply client/src/accounts/StudentSignupForm.tsx:84]: ^I think this is a good starter task or small task I might pick up later. If it comes down to it, I guess we can leave a todo in the eligibility check.

File: server/core/serializers.py [hunk 4]
```diff
@@ -21,17 +34,10 @@ def get_tokens(self, user):
...
-        instance.save()
-        return instance
-
-    class Meta:
-        model = ESPUser
-        fields = ("tokens", "username", "password")
-        extra_kwargs = {"password": {"write_only": True}}
+        profile_data = validated_data.pop("profile")
+        user = ESPUser.objects.create_user(**validated_data)
+        StudentProfile.objects.create(**profile_data, user=user)
```
[the remark this answers is not in the document]
Comment (evantey14) [reply server/core/serializers.py:39]: So, I saw that from the original code, but I think [create_user](https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.UserManager.create_user) handles this correctly? (notably `create_user()` is different than `create()`)

**Pull request merged**